ipmininet/ipnet.py

```diff
@@ -549,6 +549,47 @@ def ping6Pair(self):
            return: ploss packet loss percentage"""
         return self.pingPair(use_v4=False)
 
+   
+    def runFailurePlan(self,failure_plan):
+        """this function run a failure plan
+            the plan have to be a list of tuple with the name of routeur between
+            which the link have to be down
+            example of plan:
+            [("R1","R2"),("R8","R9"),("R3","R1")]
+        """
+        print("** Starting failure plan")
+        for link in failure_plan:
+            node1 = self.get(link[0])
```

Comment:
This can throw an exception if the node name does not exists, which at the moment abruptly terminates everything.
You need to handle this locally here, and likely perform another round of validations in the CLI handler (so your failure plan is a transaction, i.e., all failures are applied or none are).